### DISCUSSION TITLE
how can I add actions/services to child actors in react?

### DISCUSSION THREAD

**Discussion by naivefun** [statelyai/xstate#2433] [General] [answered]
when use a machine I can do `useMachine(machine, {actions: ...})`
But with actorRef there is not API to add actions to `useActor(actor, getSnapshot, {actions: ...})`

What's the correct pattern to add actions to children actors?

**Comment by christoph-fricke** [comment, marked as answer]
The important differentiation here is that `useMachine` starts an actor from the machine, and `useActor` consumes an already started actor. Likewise, a child actor spawned by a parent actor with `spawn` will already be started when it is consumed by `useActor`.
It is not possible (and does not make much sense) to change the configuration for the actions etc. after an actor is already started. Therefore, `useActor` does not accept such config object.
The other location would be the `spawn` action that created the child actor, but `spawn` does not accept such config object.

So, I guess your best bet would be to provide the configuration for the children as the second argument to the `createMachine` call for the children.

**Reply by naivefun** [reply]
IC, at least actor can invoke actions/services configured at machine level right?
